packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -624,7 +642,26 @@ export class OpenId4VcSiopVerifierService {
 
           isValid = verificationResult.verification.isValid
           reason = verificationResult.isValid ? undefined : verificationResult.error.message
-        } else if (typeof encodedPresentation === 'string') {
+        } else if (typeof encodedPresentation === 'string' && !Jwt.format.test(encodedPresentation)) {
+          if (!options.responseUri || !options.mdocGeneratedNonce) {
+            isValid = false
+            reason = 'Mdoc device response verification failed. Response uri and the mdocGeneratedNonce are not set'
+          } else {
+            const mdocDeviceResponse = MdocDeviceResponse.fromBase64Url(encodedPresentation)
+            await mdocDeviceResponse.verify(agentContext, {
+              sessionTranscriptOptions: {
+                clientId: options.audience,
+                mdocGeneratedNonce: options.mdocGeneratedNonce,
+                responseUri: options.responseUri,
+                verifierGeneratedNonce: options.nonce,
+              },
+              verificationContext: {
+                openId4VcVerificationSessionId: options.correlationId,
+              },
+            })
+            isValid = true
```

Comment:
By the way, does thi throw if it fails, or should we set the `isValid` based on the verify result?